src/server/common/model/answer/full-name/full-name.js

```diff
@@ -0,0 +1,160 @@
+import Joi from 'joi'
+import { AnswerModel } from '../answer-model.js'
+import { validateAnswerAgainstSchema } from '../validation.js'
+
+/** @import { AnswerViewModelOptions } from '../answer-model.js' */
+
+/**
+ * export @typedef {{
+ *  validation: {
+ *    firstName: {
+ *      maxLength?: { value: number, message: string },
+ *      empty?: { message: string },
+ *    },
+ *    lastName: {
+ *      maxLength?: { value: number, message: string },
+ *      empty?: { message: string },
+ *    }
+ *  }
+ * }} FullNameConfig
+ */
+
+const defaultMaxLength = 50
+
+const buildNameValidationSchema = (validation) => {
+  let stringValidation = Joi.string().trim().required()
+
+  const messages = {
+    'any.required': validation.empty?.message ?? '',
+    'string.empty': validation.empty?.message ?? '',
+    'string.max': validation.maxLength?.message ?? ''
+  }
+
+  if (validation.maxLength) {
+    stringValidation = stringValidation.max(validation.maxLength.value)
+  }
+
+  return stringValidation.messages(messages)
+}
+
+/**
+ * @param {FullNameConfig} config
+ * @returns {Joi.Schema}
+ */
+const fullNamePayloadSchema = ({ validation }) => {
+  return Joi.object({
+    firstName: buildNameValidationSchema(validation.firstName),
+    lastName: buildNameValidationSchema(validation.lastName)
+  })
+}
+
+/**
+ * export @typedef {{ firstName: string; lastName: string;}} FullNameData
```

Comment:
No need for export keyword - jsdoc seems to export all typedefs anyhow

Reply:
done